examples-gallery/plot_ball_rolling_on_spinning_disc.py

```diff
@@ -0,0 +1,594 @@
+
+**Constants**
+
+- :math:`m_b`: mass of the ball [kg]
+- :math:`r` : radius of the ball [m]
+- :math:`m_o` : mass of the observer [kg]
+- :math:`\Omega`: final rotational speed of the disc around [rad/sec]
+- :math:`\\alpha`: measure of the acceleration of the disc [1/sec]
+
+**States**
+
+- :math:`q_1, q_2, q_3`: generalized coordinates of the ball w.r.t the disc [rad]
+- :math:`u_1, u_2, u_3`: generalized angual velocities of the ball [rad/sec]
+
+**Specifieds**
+
+- :math:`t_1, t_2, t_3`: Torque applied to the ball [N * m]
+
+**Additional parameters**
+
+- :math:`N` inertial frame
+- :math:`A_1`: frame fixed to the ball
+- :math:`A_2`: frame fixed to the disc
+- :math:`O`: point fixed in N
+- :math:`C_P`: contact point of ball with disc
+- :math:`A^{o}_1`: center of mass of the ball
+- :math:`pos_{observer}`: position of the observer
+- :math:`h`: interval which opty should minimize.
+
+This example was inspired by the demonstration by Steve Mould:
+
+.. raw:: html
+
+   <iframe width="560" height="315"
+   src="https://www.youtube.com/embed/3oM7hX3UUEU?si=Q-rs1nOmikPbSBgZ"
+   title="YouTube video player" frameborder="0" allow="accelerometer; autoplay;
+   clipboard-write; encrypted-media; gyroscope;
+   picture-in-picture;
+   web-share" referrerpolicy="strict-origin-when-cross-origin"
+   allowfullscreen></iframe>
+
+
+"""
+import sympy as sm
+import sympy.physics.mechanics as me
+import numpy as np
+from opty.direct_collocation import Problem
+from opty.utils import parse_free
+from scipy.interpolate import CubicSpline
+from scipy.optimize import root
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+from matplotlib import patches
+
+# %%
+# Set up the Equations of Motion.
+#--------------------------------
+#
+# Initialize the variables.
+t = me.dynamicsymbols._t
+q1, q2, q3 = me.dynamicsymbols('q1 q2 q3')
+u1, u2, u3 = me.dynamicsymbols('u1 u2 u3')
+x, y, ux, uy = me.dynamicsymbols('x, y, ux, uy')
+t1, t2, t3 = me.dynamicsymbols('t1 t2 t3')
+
+rhs = list(sm.symbols('rhs:5'))
+
+aux = me.dynamicsymbols('auxx, auxy, auxz')
+F_r = me.dynamicsymbols('Fx, Fy, Fz')
+
+# needed to mimick the time inm opty.
+T = sm.symbols('T', cls=sm.Function)
+Tdot, Tdotdot = sm.symbols('Tdot Tdotdot')
+
+mb, mo, g, r = sm.symbols('mb, mo, g, r')
+Omega, alpha = sm.symbols('Omega, alpha')
+
+N, A1, A2 = sm.symbols('N, A1 A2', cls=me.ReferenceFrame)
+O, CP, Ao1, pos_observer = sm.symbols('O, CP, Ao1, pos_observer', cls=me.Point)
+O.set_vel(N, 0)
+
+# %%
+# Determination of the holonomic constraints.
+#
+# If the ball rotates around the x axis by an angle :math:`q_1`
+# the contact point will be move by :math:`-q_1 \cdot r` in the :math:`\hat{A2}.y`
+# direction. If the ball rotates around the y axis by an angle :math:`q_2` the
+# contact point will be move by :math:`q_2 \cdot r` in the :math:`\hat{A2}.x`
+# direction. Hence the configuration constraints are:
+#
+# - :math:`x = r \cdot q_2`
+# - :math:`y = -r \cdot q_1`
```

Comment:
I'd prefer if we didn't use `\cdot` for scalar multiplication. There isn't much reason to add a symbol for scalar multiplication, i.e. $ab$ means $a\cdotb$.